inference/huggingface/text-generation/ds-hf-compare.py

```diff
@@ -14,8 +14,13 @@
 parser.add_argument("--max_length", default=300, type=int, help="maximum tokens generated")
 parser.add_argument("--print_outputs", action='store_true', help="print generated text outputs")
 parser.add_argument("--local_rank", type=int, default=0, help="local rank")
+parser.add_argument("--use_kernel", default=True, help="enable kernel-injection")
```

Comment:
I don't believe this will work in Python as intended, please change to:
```Python
parser.add_argument("--use_kernel", action='store_true', help="enable kernel-injection")
```
See [here](https://github.com/microsoft/DeepSpeedExamples/blob/107681e647b7558c25116b277af7a90f562504d2/inference/huggingface/text-generation/arguments.py#L12) for more details.

Reply:
Hi @lekurile ! [In original code](https://github.com/microsoft/DeepSpeedExamples/blob/master/inference/huggingface/text-generation/ds-hf-compare.py#L88), the kernel injection is enabled by default. We want to maintain the original behavior of the code before and after modification, and the added argument is only used to disable KI when CPUs are applied. However, based on your suggestion, it seems we need to add the additional argument to keep kernel injection enabled. Is this appropriate? Please reconsider and give me feedback. Thanks a lot.

Reply:
Hi @foin6, we don't necessarily have to maintain the original behavior regarding kernel injection. In fact, as I linked previously [here](https://github.com/microsoft/DeepSpeedExamples/blob/107681e647b7558c25116b277af7a90f562504d2/inference/huggingface/text-generation/arguments.py#L12), we disable it by default and require an explicit argument to enable it. I think it's perfectly reasonable to add a `use_kernel` argument that needs to be passed to enable kernel injection for this script as it will align closer to other existing inference scripts.

Regardless of this point, the Python for the argument currently written won't actually work when providing `True/False` and will have to be changed in order to work, hence my suggestion to change the code.

Reply:
ok, Thanks for your explanation!

Reply:
Solved